homeassistant/components/nextbus/__init__.py

```diff
@@ -1 +1,32 @@
-"""NextBus sensor."""
+"""NextBus platform."""
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_AGENCY, DOMAIN
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up platforms for NextBus."""
+    entry_agency = entry.data.get(CONF_AGENCY)
+    if not entry_agency:
+        return False
```

Comment:
This should not be necessary.
It should instead be checked as part of the config flow or the import.

Reply:
Yes, you should remove the redundant check here.
The config flow has the agency as compulsory.

Reply:
Got it. I'll do similar between steps in the config flow then. Since Agency is compulsory, I'll not check for an invalid value and rely on the Schema to do that for me.